specs/WebResourceRequestedWorkers.md

```diff
@@ -0,0 +1,210 @@
+  /// one CoreWebView2 to avoid handling the same WebResourceRequested
+  /// event multiple times.
+  ///
+  /// | URI Filter String | Request URI | Match | Notes |
+  /// | ---- | ---- | ---- | ---- |
+  /// | `*` | `https://contoso.com/a/b/c` | Yes | A single * will match all URIs |
+  /// | `*://contoso.com/*` | `https://contoso.com/a/b/c` | Yes | Matches everything in contoso.com across all schemes |
+  /// | `*://contoso.com/*` | `https://example.com/?https://contoso.com/` | Yes | But also matches a URI with just the same text anywhere in the URI |
+  /// | `example` | `https://contoso.com/example` | No | The filter does not perform partial matches |
+  /// | `*example` | `https://contoso.com/example` | Yes | The filter matches across URI parts  |
+  /// | `*example` | `https://contoso.com/path/?example` | Yes | The filter matches across URI parts |
+  /// | `*example` | `https://contoso.com/path/?query#example` | No | The filter is matched against the URI with no fragment |
+  /// | `*example` | `https://example` | No | The URI is normalized before filter matching so the actual URI used for comparison is `https://example.com/` |
+  /// | `*example/` | `https://example` | Yes | Just like above, but this time the filter ends with a / just like the normalized URI |
+  /// | `https://xn--qei.example/` | `https://&#x2764;.example/` | Yes | Non-ASCII hostnames are normalized to punycode before wildcard comparison |
+  /// | `https://&#x2764;.example/` | `https://xn--qei.example/` | No | Non-ASCII hostnames are normalized to punycode before wildcard comparison |
+  HRESULT AddWebResourceRequestedFilterWithRequestSourceKinds(
+    [in] LPCWSTR const uri,
+    [in] COREWEBVIEW2_WEB_RESOURCE_CONTEXT const resourceContext,
+    [in] COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS const requestSourceKinds);
+
+  /// Removes a matching WebResource filter that was previously added for the
+  /// `WebResourceRequested` event.  If the same filter was added multiple
+  /// times, then it must be removed as many times as it was added for the
+  /// removal to be effective.  Returns `E_INVALIDARG` for a filter that was
+  /// never added.
+  HRESULT RemoveWebResourceRequestedFilterWithRequestSourceKinds(
+      [in] LPCWSTR const uri,
+      [in] COREWEBVIEW2_WEB_RESOURCE_CONTEXT const resourceContext,
+      [in] COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS const requestSourceKinds);
+ }
+
+/// Event args for the `WebResourceRequested` event.
+[uuid(572f38f9-c317-4b6c-8dd2-c8b894779bbb), object, pointer_default(unique)]
+interface ICoreWebView2WebResourceRequestedEventArgs2 : ICoreWebView2WebResourceRequestedEventArgs {
+  /// The web resource request source.
+  [propget] HRESULT RequestSourceKinds(
+  [out, retval] COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS* requestSourceKinds);
+}
+```
+
+## .NET and WinRT
+```c# (but really MIDL3)
+namespace Microsoft.Web.WebView2.Core
+{
+    [flags]
+    enum CoreWebView2WebResourceRequestSourceKinds
+    {
```

Comment:
Missing None=0x0.  WinRT API guidelines are to be explicit about it.  Any reason not to include that? (API comment says we already check for zero, named enum seems clearer)

Reply:
None would mean that you don't want the event to ever be raised?